File: backend/app/api/v1/endpoints/health.py
from typing import Any, Dict
from fastapi import APIRouter
from app.core.config import settings
from app.core.database import check_db_health

router = APIRouter()


@router.get("/health", response_model=Dict[str, Any], summary="Health Check")
async def health_check() -> Dict[str, Any]:
    db_connected = await check_db_health()
    return {
        "status": "ok" if db_connected else "degraded",
        "service": settings.PROJECT_NAME,
        "environment": settings.ENVIRONMENT,
        "database": "connected" if db_connected else "disconnected",
    }

File: backend_rust/src/api/v1/health.rs
use axum::{
    extract::State,
    routing::get,
    Json, Router,
};
use serde::Serialize;
use sqlx::PgPool;
use crate::config::Settings;
use crate::db::check_db_health;

#[derive(Serialize)]
pub struct HealthResponse {
    pub status: String,
    pub service: String,
    pub environment: String,
    pub database: String,
}

pub fn router() -> Router<crate::AppState> {
    Router::new()
        .route("/health", get(health_check))
}

async fn health_check(
    State(pool): State<PgPool>,
    State(settings): State<Settings>,
) -> Json<HealthResponse> {
    let db_connected = check_db_health(&pool).await;
    Json(HealthResponse {
        status: if db_connected { "ok" } else { "degraded" }.to_string(),
        service: settings.project_name,
        environment: settings.environment,
        database: if db_connected { "connected" } else { "disconnected" }.to_string(),
    })
}